src/servers/http_server.cc

```diff
@@ -321,280 +142,291 @@ class HTTPAPIServer : public HTTPServerImpl {
-          1) {
-        return TRTSERVER_ErrorNew(
-            TRTSERVER_ERROR_INTERNAL,
-            std::string(
-                "failed to reserve " + std::to_string(byte_size) +
-                " bytes in output tensor buffer")
-                .c_str());
-      }
+  if (status != 0)
+    return;
 
-      if (output_iovec.iov_len < byte_size) {
-        return TRTSERVER_ErrorNew(
-            TRTSERVER_ERROR_INTERNAL,
-            std::string(
-                "reserved " + std::to_string(output_iovec.iov_len) +
-                " bytes in output tensor buffer, need " +
-                std::to_string(byte_size))
-                .c_str());
-      }
+  conn = reinterpret_cast<uv_tcp_t*>(h2o_mem_alloc(sizeof(*conn)));
+  uv_tcp_init(listener->loop, conn);
 
-      output_iovec.iov_len = byte_size;
-      *buffer = output_iovec.iov_base;
-
-      // Immediately commit the buffer space. We are relying on evbuffer
-      // not to relocate this space. Because we request a contiguous
-      // chunk every time (above by allowing only a single entry in
-      // output_iovec), this seems to be a valid assumption.
-      if (evbuffer_commit_space(evhttp_buffer, &output_iovec, 1) != 0) {
-        *buffer = nullptr;
-        return TRTSERVER_ErrorNew(
-            TRTSERVER_ERROR_INTERNAL,
-            "failed to commit output tensors to output buffer");
-      }
-    }
+  if (uv_accept(listener, (uv_stream_t*)conn) != 0) {
+    uv_close((uv_handle_t*)conn, (uv_close_cb)free);
+    return;
   }
 
-  LOG_VERBOSE(1) << "HTTP allocation: '" << tensor_name
-                 << "', size: " << byte_size << ", addr: " << *buffer;
-
-  return nullptr;  // Success
+  sock = h2o_uv_socket_create((uv_stream_t*)conn, (uv_close_cb)free);
+  h2o_accept(&http_server->accept_ctx, sock);
 }
 
-TRTSERVER_Error*
-HTTPAPIServer::ResponseRelease(
-    TRTSERVER_ResponseAllocator* allocator, void* buffer, void* buffer_userp,
-    size_t byte_size, TRTSERVER_Memory_Type memory_type, int64_t memory_type_id)
+int
+HTTPAPIServer::create_listener(int32_t port)
 {
-  LOG_VERBOSE(1) << "HTTP release: "
-                 << "size " << byte_size << ", addr " << buffer;
+  struct sockaddr_in addr;
+  int r;
 
-  // Don't do anything when releasing a buffer since ResponseAlloc
-  // wrote directly into the response ebvuffer.
-  return nullptr;  // Success
+  uv_tcp_init(ctx.loop, &listener);
+  uv_ip4_addr("0.0.0.0", port, &addr);
+  if ((r = uv_tcp_bind(&listener, (struct sockaddr*)&addr, 0)) != 0) {
+    LOG_VERBOSE(1) << stderr << "uv_tcp_bind:" << uv_strerror(r);
+    goto Error;
+  }
+  if ((r = uv_listen((uv_stream_t*)&listener, 128, on_accept)) != 0) {
+    LOG_VERBOSE(1) << stderr << "uv_listen:" << uv_strerror(r);
+    goto Error;
+  }
+
+  listener.data = this;
+
+  return 0;
+Error:
+  uv_close((uv_handle_t*)&listener, NULL);
+  return r;
 }
 
-void
-HTTPAPIServer::Handle(evhtp_request_t* req)
+int
+HTTPAPIServer::setup_ssl(
```

Comment:
Can we delete ssl for now (or #ifdef this out if it will be hard to find this same information later)

Reply:
Put it in an ifdef